### ISSUE TITLE
Different accuracy testing on the fly and testing with snapshot models

### ISSUE THREAD

**Issue by weeoooweeooo** [facebook/C3D#144]
Hello, 

A really interesting project you have going here. I've been experimenting with it recently, using the package's example models, without any issue. When fine-tuning the pre-trained sport1m model with my own data however, I'm now seeing something I can't explain:

I've set up for testing on the fly while fine-tuning on a subset of my data. Everything appears to be proceeding smoothly, though accuracy reported in this manner differs from that reported testing separately with intermediate snapshot models created. Accuracy reported testing on the fly appears to be consistently lower by a few percentage points, though is setup pointing to the same test prototxt used with the snapshots. Why would this be the case?

Shuffling is set up for testing, though the test set is regrettably very small (n=80). Is this related?

Thank you in advance

**Comment by dutran** [comment]
testing on the fly sometime tested on a small number of mini-batches, while testing at the end is over all test set.
